src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -281,6 +297,76 @@ public bool IsSolutionOpen
             }
         }
 
+        public async Task<Tuple<bool, bool>> CheckSolutionUIPreConditionsAsync()
```

Comment:
This is the method declaration in `IVsSolutionManager`

/// <summary>
        /// Check for all the pre-conditions before showing manager ui at solution level. And
        /// return tuple of two items. Item1 represents if solution is available to work. And
        /// item2 represents if solution has any project which NuGet supports.
        /// </summary>
        Task<Tuple<bool, bool>> CheckSolutionUIPreConditionsAsync();